### PR TITLE
Fix support to transactions with no expense nor order

### PR DESCRIPTION
**Pull request by kewitz**
Related to https://github.com/opencollective/opencollective/issues/3385
Requires https://github.com/opencollective/opencollective-api/pull/4324

### CODE REVIEW

File: components/transactions/TransactionDetails.js [hunk 1]
```diff
@@ -67,28 +67,30 @@ const TransactionDetails = ({
...
+            <Box>
+              <DetailTitle>
+                <FormattedMessage id="Member.Role.FISCAL_HOST" defaultMessage="Fiscal Host" />
+              </DetailTitle>
+              <DetailDescription>
+                <LinkCollective collective={toAccount.host} />
+              </DetailDescription>
+            </Box>
+          )}
+          {order?.paymentMethod && (
```
Comment (znarf) [inline components/transactions/TransactionDetails.js:82]: `paymentMethod` should be read at the transaction level I think

File: components/transactions/TransactionDetails.js [hunk 2]
```diff
@@ -67,28 +67,30 @@ const TransactionDetails = ({
...
-            <DetailTitle>
-              <FormattedMessage id="PaidWith" defaultMessage="Paid With" />
-            </DetailTitle>
-            <DetailDescription>
-              <PaymentMethodTypeWithIcon type={order.paymentMethod.type} fontSize={11} iconSize={16} />
-            </DetailDescription>
-          </Box>
-        )}
-      </Flex>
+      {toAccount.host && order?.paymentMethod && (
```
Comment (znarf) [inline components/transactions/TransactionDetails.js:70]: I'm wondering why we should skip this if no `paymentMethod`.

Comment (kewitz) [reply components/transactions/TransactionDetails.js:70]: Good catch, this should be an `||`.

File: components/transactions/TransactionDetails.js (hunk 1 above)
Comment (kewitz) [reply components/transactions/TransactionDetails.js:82]: True. I'll implement the missing resolver and it altogether.

**Pull request merged**